meson.build

```
@@ -390,6 +390,21 @@ possible_common_cc_flags = [
 
 c_args = get_option('c_args')
 
+# Our json library does not support -ffinite-math-only, which is enabled by -Ofast or -ffast-math.
+# Refuse to build in that case.
+have = false
+foreach a : c_args
+        if a == '-Ofast' or a == '-ffast-math' or a == '-ffinite-math-only'
```

```suggestion
        if a in ['-Ofast', '-ffast-math', '-ffinite-math-only']
```

This also fails to detect a later override with `-O2` for example. (Personally, I think this loop is just overkill.)